faunadb-android/src/test/java/com/faunadb/client/AndroidClientSpec.java

```diff
@@ -51,4 +68,60 @@ public void shouldAuthenticateSession() throws Exception {
     assertThat(identified.to(BOOLEAN).get(), is(false));
   }
 
-}
\ No newline at end of file
+  @Before
+  public void setUpClient() throws Exception {
+    if (rootClient != null)
+      return;
+
+    rootClient = createFaunaClient(ROOT_TOKEN);
+
+    client = transform(
+      transformAsync(
+        setupDatabase(DB_REF, DB_NAME),
```

Comment:
I don't think the `DslSpec` should create the database. I think it should be a separated class.

To get around the need for a client, this separated class could return a `Expr`. Something like `public static Expr createDatabase(String name)`. Each driver test could them call those methods to create the database and its key.

Therefore, the before method could be moved up to the `DslSpec`. `DslSpec` should only deal with the data it needs to test the dsl, like creating classes and instances.

That was my idea.